library/core/src/str/validations.rs

```diff
@@ -34,8 +34,21 @@ pub(super) const fn utf8_is_cont_byte(byte: u8) -> bool {
 #[unstable(feature = "str_internals", issue = "none")]
 #[inline]
 pub unsafe fn next_code_point<'a, I: Iterator<Item = &'a u8>>(bytes: &mut I) -> Option<u32> {
+    // SAFETY: `bytes` must produce a valid UTF-8-like (UTF-8 or WTF-8) string
+    unsafe { next_code_point_val(&mut bytes.copied()) }
```

Comment:
considering that `next_code_point` is unstable, it's probably better to change `next_code_point` to take `Iterator<Item = u8>` directly. Any use can get fixed by using `copied()`.

Reply:
I had considered that, but I wanted to minimize my impact, mostly because the process of running the unit tests takes a really long time.  I can create a second branch that alters `next_code_point` to see if that version is preferred by folks with merge authority.

Reply:
https://github.com/rust-lang/rust/pull/96019 is a variant that modifies `next_code_point`.